lib/reverse_sentence.rb

```diff
@@ -1,6 +1,31 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
```

Comment:
Note:  by doing `my_sentence.clone` you make a copy of the original sentence, so you have minimum space complexity of O(n).

There is a way to do this with O(n) time complexity and O(1) space complexity by avoiding the creation of an external array.  I encourage you to think about how this might be accomplished and bring questions to class.